backend/infrahub/message_bus/operations/schema/validator.py

```diff
@@ -41,3 +45,43 @@ async def path(message: SchemaValidatorPath, service: InfrahubServices) -> None:
                 )
             )
             await service.reply(message=response, initiator=message)
+
+
+def generate_task_name() -> str:
+    task_name = task_run.task_name
+    message: SchemaValidatorPathData = task_run.parameters["message"]
+    return f"{task_name}-{message.branch.name}-{message.constraint_name}"
+
+
+@task(
+    task_run_name=generate_task_name,
+    description="Validate if a given migration is compatible with the existing data",
+    retries=3,
+)
+async def schema_path_validate(message: SchemaValidatorPathData) -> SchemaValidatorPathResponseData:
```

Comment:
it looks like `schema_path_validate` and `path` above could be refactored to share a `_do_validate` (or better name) method

same for `schema_path_migrate` above

Reply:
Ultimately the goal is for `schema_path_validate` to completely replace `path` so haven't looked too closely on how to refactor the code.
I need to fix a circular dependencies around SchemaBranch first :/